test/metabase/cmd/dump_to_h2_test.clj

```diff
@@ -67,43 +66,43 @@
 (deftest dump-to-h2-dump-plaintext-test
   (testing "dump-to-h2 --dump-plaintext"
     (let [h2-fixture-db-file  @cmd.test-util/fixture-db-file-path
-          h2-file-plaintext   (format "/tmp/out-%s.db" (mt/random-name))
-          h2-file-enc         (format "/tmp/out-%s.db" (mt/random-name))
-          h2-file-default-enc (format "/tmp/out-%s.db" (mt/random-name))
           db-name             (str "test_" (mt/random-name))]
-      (mt/test-drivers #{:h2 :postgres :mysql}
-        (with-redefs [i18n.impl/site-locale-from-setting-fn (atom (constantly false))]
-          (binding [setting/*disable-cache*    true
-                    mdb.connection/*db-type*   driver/*driver*
-                    mdb.connection/*jdbc-spec* (persistent-jdbcspec driver/*driver* db-name)
-                    db/*db-connection*         (persistent-jdbcspec driver/*driver* db-name)
-                    db/*quoting-style*         driver/*driver*]
-            (when-not (= driver/*driver* :h2)
-              (tx/create-db! driver/*driver* {:database-name db-name}))
-            (load-from-h2/load-from-h2! h2-fixture-db-file)
-            (eu/with-secret-key "89ulvIGoiYw6mNELuOoEZphQafnF/zYe+3vT+v70D1A="
-              (db/insert! Setting {:key "my-site-admin", :value "baz"})
-              (db/update! Database 1 {:details "{\"db\":\"/tmp/test.db\"}"})
-              (dump-to-h2/dump-to-h2! h2-file-plaintext {:dump-plaintext? true})
-              (dump-to-h2/dump-to-h2! h2-file-enc {:dump-plaintext? false})
-              (dump-to-h2/dump-to-h2! h2-file-default-enc))
+      (mt/with-temp-file [h2-file-plaintext (format "/tmp/out-%s.db" (mt/random-name))]
```

Comment:
Can this be made to take varargs, similar to `mt/with-temp*` to prevent this nesting?